### PR TITLE
Copy references

### PR DESCRIPTION
**Pull request by dantleech**
NOTE: this PR is based on the numerical sorting PR (#226) which would need to be merged first. 

If that PR is rejected I will need to rebase and update this PR, probably better to delay reviewing this until that is merged.

For changes see the second commit in this PR.

### Depends

- https://github.com/phpcr/phpcr-api-tests/pull/148

### CODE REVIEW

**Comment by lsmith77** [comment]
please rebase

File: src/Jackalope/Transport/DoctrineDBAL/Client.php [hunk 1]
```diff
@@ -589,9 +597,24 @@ public function copyNode($srcAbsPath, $dstAbsPath, $srcWorkspace = null)
                 'stringDom' => $stringDom,
                 'numericalDom' => $numericalDom
             );
+
+            $xpath = new \DOMXpath($stringDom);
+            $referenceEls = $xpath->query('.//sv:property[@sv:type="reference"]');
+
+            foreach ($referenceEls as $referenceEl) {
+                if (array_key_exists($referenceEl->nodeValue, $resultSetUuids)) {
```
Comment (dantleech) [inline src/Jackalope/Transport/DoctrineDBAL/Client.php]: Should use `isset` here I think.

**Comment by dantleech** [comment]
Rebased and updated the related phpcr-api-tests PR.

**Comment by dantleech** [comment]
@lsmith77 the branch is already up-to-date here

**Comment by lsmith77** [comment]
yes .. but if you rebase to master then the tests should pass

**Comment by dantleech** [comment]
I will restart the tests .. but I can't rebase on master because there is nothing to rebase on.

**Comment by dantleech** [comment]
All green :)

**Comment by dbu** [comment]
i wrote a comment on https://github.com/phpcr/phpcr-api-tests/pull/148 - can you check that? would love to merge the tests first, then restart this build one more time, just to be sure.

**Comment by dbu** [comment]
merged the tests and restarted the build

**Comment by dantleech** [comment]
Last test run failed with the random problem (cannot delete node `/../idExample`) @dbu mentioned a few days ago.

**Comment by dantleech** [comment]
Build is green now however ...

**Pull request merged**